File: examples/python/create_secret.py
#!/usr/bin/env python3
import base64
import json
import os
import sys
import urllib.error
import urllib.request


def env(name, default=None):
    value = os.environ.get(name, default)
    if value is None or value == "":
        print(f"{name} is required", file=sys.stderr)
        sys.exit(2)
    return value


base_url = env("SECURESHARE_BASE_URL", "http://localhost:8080").rstrip("/")
client_id = env("SECURESHARE_CLIENT_ID")
client_secret = env("SECURESHARE_CLIENT_SECRET")

payload = {
    "title": "Example login",
    "expires_in_seconds": 3600,
    "payload": {
        "type": "structured",
        "fields": [
            {"name": "username", "label": "Username", "value": "example-user", "sensitive": False, "multiline": False},
            {"name": "password", "label": "Password", "value": "example-password", "sensitive": True, "multiline": False},
        ],
    },
}

raw = json.dumps(payload).encode("utf-8")
token = base64.b64encode(f"{client_id}:{client_secret}".encode("utf-8")).decode("ascii")
request = urllib.request.Request(
    f"{base_url}/api/v1/secret-links",
    data=raw,
    headers={
        "Authorization": f"Basic {token}",
        "Content-Type": "application/json",
    },
    method="POST",
)

try:
    with urllib.request.urlopen(request, timeout=15) as response:
        body = response.read()
except urllib.error.HTTPError as exc:
    print(f"SecureShare returned HTTP {exc.code}", file=sys.stderr)
    sys.exit(1)

print(json.loads(body)["url"])

File: examples/javascript/create-secret.mjs
const baseURL = (process.env.SECURESHARE_BASE_URL || "http://localhost:8080").replace(/\/$/, "");
const clientID = process.env.SECURESHARE_CLIENT_ID;
const clientSecret = process.env.SECURESHARE_CLIENT_SECRET;

if (!clientID || !clientSecret) {
  console.error("SECURESHARE_CLIENT_ID and SECURESHARE_CLIENT_SECRET are required");
  process.exit(2);
}

const payload = {
  title: "Example login",
  expires_in_seconds: 3600,
  payload: {
    type: "structured",
    fields: [
      { name: "username", label: "Username", value: "example-user", sensitive: false, multiline: false },
      { name: "password", label: "Password", value: "example-password", sensitive: true, multiline: false },
    ],
  },
};

const controller = new AbortController();
const timeout = setTimeout(() => controller.abort(), 15000);

try {
  const response = await fetch(`${baseURL}/api/v1/secret-links`, {
    method: "POST",
    headers: {
      Authorization: `Basic ${Buffer.from(`${clientID}:${clientSecret}`).toString("base64")}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
    signal: controller.signal,
  });

  if (!response.ok) {
    console.error(`SecureShare returned HTTP ${response.status}`);
    process.exit(1);
  }

  const body = await response.json();
  console.log(body.url);
} finally {
  clearTimeout(timeout);
}
